File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        int[] numbers = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int[] prefixSum = new int[n+1];
        prefixSum[0] = 0;
        for (int i=0; i<n; i++) {
            prefixSum[i+1] = prefixSum[i] + numbers[i];
        }

        for (int q=0; q<m; q++) {
            st = new StringTokenizer(br.readLine());
            int i = Integer.parseInt(st.nextToken());
            int j = Integer.parseInt(st.nextToken());
            System.out.println(prefixSum[j] - prefixSum[i-1]);
        }
    }
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.kt
/**
 * p11659 구간 합 구하기4
 */
fun main() {
    val (n, m) = readln().split(" ").map { it.toInt() }
    val numbers = readln().split(" ").map { it.toInt() }

    val prefixSum = IntArray(n+1)
    prefixSum[0] = 0
    for (i in 1 .. n) {
        prefixSum[i] = prefixSum[i-1] + numbers[i-1]
    }

    repeat(m) {
        val (i, j) = readln().split(" ").map { it.toInt() }
        println(prefixSum[j] - prefixSum[i-1])
    }
}